### ISSUE TITLE
Files resolved via paths confg don't get source maps.

### ISSUE THREAD

**Issue by wwalser** [scalableminds/amd-optimize#56]
I'm migrating a project that was using the require optimizer to amd-optimize. I've been struggling with a build that wasn't getting sourcemaps output and have tracked the problem down to files which get resolved via ```paths``` configuration. 

My config looks something like this:

    var amdConfig = {
        "baseUrl": "./target/js",
        "preserveFiles": true,
        "paths": {
            "class": "/lib/class-amd",
            "passwordstrength": "/lib/passwordstrength-amd",
            "jquery": "empty:",
        }
    };

And my task looks more or less like this:

    return gulp.src('target/js/**/*.js', { base: amdConfig.baseUrl })
        .pipe(sourcemaps.init())
        .pipe(amdOptimize("project-name/bootstrap", amdConfig))
        .pipe(concat('bootstrap.js'))
        .pipe(sourcemaps.write())
        .pipe(gulp.dest('./target/js/built'));

I found the problem by removing the concat line and noting that all files appeared to have sourcemaps. This lead me to thinking that perhaps a few files didn't so I inserted a debug log into gulp-sourcemaps to let me know if this was the case. It turns out that both class and password strength don't have sourcemaps associated. This leads to no files having sourcemaps after concat.

If I write a fix I'll PR it here. Just raising the bug here now in case you're more prepared to fix since you are more familiar with the code.

Thanks for the great project :).

**Comment by wwalser** [comment]
mm, upon reading the code

    # TODO: Handle shimmed, mapped and relative deps

Is probably relevant.

**Comment by wwalser** [comment]
I tracked down the culprit. When [index's module.exports.loader](https://github.com/scalableminds/amd-optimize/blob/master/src/index.coffee#L221) get's used instead of the other methods of loading a file, source maps don't exist on the loaded files as they haven't gone through ```sourcemaps.init()```. Removing the leading ```/``` from my paths declarations resolved the issue.

Seem like an issue with my configuration. Primarily that paths wasn't being used as expected. I believe I had changed them to use a leading slash because using a leading ```./``` causes problems in how the file is looked up inside ```trace```. If a path starts with a dot, it ends up assuming the path is relative to the module that depends on it.

I'll close this issue now.
